File: regexlib/2021-5-15/javascript_test_file/regexlib_2403.js
/* 2403
 * actionID=&url=&load_frameset=1&autologin=0&anchor_string=&server_key=imap&imapuser=(.*)&pass=(.*)&new_lang=pt_BR&select_view=imp
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"actionID=&url=&load_frameset=1&autologin=0&anchor_string=&server_key=imap&imapuser=&pass="*256+"! _1SLQ_2"
 */
var REGEX = /actionID=&url=&load_frameset=1&autologin=0&anchor_string=&server_key=imap&imapuser=(.*)&pass=(.*)&new_lang=pt_BR&select_view=imp/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'actionID=&url=&load_frameset=1&autologin=0&anchor_string=&server_key=imap&imapuser=&pass='.repeat(i*1) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_2403.py
# 2403
# actionID=&url=&load_frameset=1&autologin=0&anchor_string=&server_key=imap&imapuser=(.*)&pass=(.*)&new_lang=pt_BR&select_view=imp
# EXPONENT
# nums:4
# EXPONENT AttackString:""+"actionID=&url=&load_frameset=1&autologin=0&anchor_string=&server_key=imap&imapuser=&pass="*256+"! _1SLQ_2"

import re
from time import perf_counter

regex = """actionID=&url=&load_frameset=1&autologin=0&anchor_string=&server_key=imap&imapuser=(.*)&pass=(.*)&new_lang=pt_BR&select_view=imp"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "actionID=&url=&load_frameset=1&autologin=0&anchor_string=&server_key=imap&imapuser=&pass=" * i * 1 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")